pkg/client/cli/cliutil/exec_unix.go

```diff
@@ -0,0 +1,74 @@
+// +build !windows
+
+package cliutil
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+
+	//nolint:depguard // Because we won't ever .Wait() for the process and we'd turn off
+	// logging, using dexec would just be extra overhead.
+	"os/exec"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/datawire/dlib/dcontext"
+	"github.com/datawire/dlib/dexec"
+	"github.com/telepresenceio/telepresence/v2/pkg/client/logging"
+	"github.com/telepresenceio/telepresence/v2/pkg/proc"
+)
+
+func background(ctx context.Context, exe string, args []string) error {
+	// The context should not kill it if cancelled
+	ctx = dcontext.WithoutCancel(ctx)
+	cmd := exec.CommandContext(ctx, exe, args...)
+
+	// Ensure that the processes uses a process group of its own to prevent
+	// it getting affected by <ctrl-c> in the terminal
+	cmd.SysProcAttr = &unix.SysProcAttr{Setpgid: true}
+
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("%s: %w", logging.ShellString(exe, args), err)
+	}
+	if err := cmd.Process.Release(); err != nil {
+		return fmt.Errorf("%s: %w", logging.ShellString(exe, args), err)
+	}
+	return nil
+}
+
+func backgroundAsRoot(ctx context.Context, exe string, args []string) error {
+	if !proc.IsAdmin() {
+		// If we're going to be prompting for the `sudo` password, we want to first provide
+		// the user with some info about exactly what we're prompting for.  We don't want to
+		// use `sudo`'s `--prompt` flag for this because (1) we don't want it to be
+		// re-displayed if they typo their password, and (2) it might be ignored anyway
+		// depending on `passprompt_override` in `/etc/sudoers`.  So we'll do a pre-flight
+		// `sudo --non-interactive true` to decide whether to display it.
+		needPwCmd := dexec.CommandContext(ctx, "sudo", "--non-interactive", "true")
+		needPwCmd.DisableLogging = true
+		if err := needPwCmd.Run(); err != nil {
+			fmt.Printf("Need root privileges to run: %s\n", logging.ShellString(exe, args))
+			// `sudo` won't be able to read the password from the terminal when we run
+			// it with Setpgid=true, so do a pre-flight `sudo true` to read the
+			// password, and then enforce that being re-used by passing
+			// `--non-interactive`.
+			pwCmd := dexec.CommandContext(ctx, "sudo", "true")
+			pwCmd.DisableLogging = true
+			if err := pwCmd.Run(); err != nil {
+				return err
+			}
+		}
+		args = append([]string{"--non-interactive", "--preserve-env", exe}, args...)
+		exe = "sudo"
+	}
+	return Background(ctx, exe, args)
```

Comment:
I'm not a fan of this business with having `exe` be separate than just `args[0]`.

Reply:
This is the only comment I didn't really address for two reasons:

1. I think it's pretty reasonable, especially in code like this where the executable is being modified. IMHO it'd look a little bit weird to do `args[0] = "sudo"` instead of `exe = "sudo"`
2. This is consistent with the `dexec.Command*` APIs